homework6/task1/main.c

```diff
@@ -0,0 +1,168 @@
+#include "stack.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+typedef enum ErrorCode
+{
+    ok = 0,
+    divisionByZeroError = -1,
+    memoryAllocationError = -2,
+    operandFindingError = -3,
+    incorrectPostfixExpressionError = -4,
+    invalidOperatorError = -5
+} ErrorCode;
+
+#define EXPRESSION_LENGTH 100
+
+int calculatePostfixExpression(char *expression, ErrorCode* errorCode)
+{
+    *errorCode = ok;
+
+    Stack* stack = stackCreate();
+    if (stack == NULL)
+    {
+        *errorCode = memoryAllocationError;
+        return 0;
+    }
+
+    int result = 0;
+
+    for (size_t i = 0; expression[i] != '\0'; ++i)
+    {
+        int operand1 = 0;
+        int operand2 = 0;
+        char symbol = expression[i];
+
+        if (symbol == ' ' || symbol == '\n')
+        {
+            continue;
+        }
+
+        if (symbol >= '0' && symbol <= '9')
+        {
+            stackPush(stack, symbol - '0');
+        }
+        else
+        {
+            operand2 = 0;
+            operand1 = 0;
+
+            if (stackPop(stack, &operand2) != stackOk || stackPop(stack, &operand1) != stackOk)
+            {
+                *errorCode = operandFindingError;
+                stackFree(&stack);
+                return 0;
+            }
+
+            switch (symbol)
+            {
+            case '+':
+            {
+                result = operand1 + operand2;
+                break;
+            }
+            case '-':
+            {
+                result = operand1 - operand2;
+                break;
+            }
+            case '*':
+            {
+                result = operand1 * operand2;
+                break;
+            }
+            case '/':
+            {
+                if (operand2 != 0)
+                {
+                    result = operand1 / operand2;
+                }
+                else
+                {
+                    *errorCode = divisionByZeroError;
+                    stackFree(&stack);
+                    return 0;
+                }
+                break;
+            }
+            default:
+                *errorCode = invalidOperatorError;
+                return 0;
+                break;
```

Comment:
```suggestion
            default:
            {
                *errorCode = invalidOperatorError;
                return 0;
                break;
            }
```
Если делаете так, то стоит делать во всём блоке switch